lld/MachO/Driver.cpp

```diff
@@ -1169,7 +1169,8 @@ static void createFiles(const InputArgList &args) {
 
     switch (opt.getID()) {
     case OPT_INPUT:
-      addFile(rerootPath(arg->getValue()), LoadType::CommandLine, isLazy);
+      addFile(rerootPath(arg->getValue()), LoadType::CommandLine,
+              isLazy && !config->allLoad);
```

Comment:
It's a bit confusing to override `isLazy` for each case. Instead, can we simply not set `isLazy` below in `case OPT_start_lib:` with `config->allLoad` ?

Reply:
Got it. Using another variable seems clearer to me.
```
case OPT_start_lib:
  if (inLib)
    error("nested --start-lib");
  inLib = true;
  if (!config->allLoad)
    isLazy = true;
  break;
case OPT_end_lib:
  if (!inLib)
    error("stray --end-lib");
  inLib = false;
  if (!config->allLoad)
    isLazy = false;
  break;
```

Reply:
What if adding on line ~278:
```
bool loadLazily = isLazy && !config->allLoad;
```
And updating lines 370 + 380 to use `loadLazily` ?

Reply:
Added a combination of the two, I like having an `inLib` variable to make it clear there's a difference between being inside `--start-lib` and actually loading lazily.